apps/federatedfilesharing/lib/FedShareManager.php

```diff
@@ -328,8 +320,9 @@ public function updateOcmPermissions(IShare $share, $ocmPermissions) {
 	 *
 	 * @return void
 	 */
-	public function updatePermissions(IShare $share, $permissions) {
-		if ($share->getPermissions() !== $permissions) {
+	public function updatePermissions(IShare $share, int $permissions): void {
+		# permissions can only be reduced but not upgraded
+		if (Permissions::isNewPermissionHigher($share->getPermissions(), $permissions)) {
```

Comment:
```suggestion
	public function updatePermissions(IShare $share, int $newPermissions): void {
		# permissions can only be reduced but not upgraded
		if (Permissions::isNewPermissionLower($share->getPermissions(), $newPermissions)) {
```

We only want to allow the permissions to be adjusted lower. (And if they are the same, then there is nothing to do)

Reply:
See #40829